### PR TITLE
fix: Remove warnings for S3 bucket server side encryption configuration & acl

### PR DESCRIPTION
**Pull request by kirzharov**
# What

Move S3 bucket server side encryption configuration and bucket acl to separate resources to remove warnings after updating Tecton deployment higher than `3.75.0` version for `hashicorp/aws`.

# Why

After we updated our Tecton deployment, we noticed a couple of warnings:

1. About S3 bucket server side encryption configuration:
```
Warning: Argument is deprecated

  on .terraform/modules/tecton/deployment/buckets.tf line 1, in resource "aws_s3_bucket" "tecton":
   1: resource "aws_s3_bucket" "tecton" {

Use the aws_s3_bucket_server_side_encryption_configuration resource instead
```

2. About S3 bucket acl:
```
Warning: Argument is deprecated

  on .terraform/modules/tecton/deployment/buckets.tf line 3, in resource "aws_s3_bucket" "tecton":
   3:   acl    = "private"

Use the aws_s3_bucket_acl resource instead
```

### CODE REVIEW

File: deployment/versions.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.75.0"
```
Comment (scottames) [inline deployment/versions.tf:5]:
```suggestion
      version = "~> 3"
```

Comment (scottames) [reply deployment/versions.tf:5]: Thanks for this @kirzharov! Would you mind pinning the AWS provider version to 3.x instead?

**Review by scottames — CHANGES_REQUESTED** [review]

File: deployment/versions.tf (hunk 1 above)
Comment (kirzharov) [reply deployment/versions.tf]: Hi @scottames! I switched to 3.x version, but we need keep in mind that AWS resources `aws_s3_bucket_server_side_encryption_configuration` and `aws_s3_bucket_acl` have been added with `3.75.0` version ([https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3750-march-18-2022](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3750-march-18-2022))

**Review by scottames — APPROVED** [review]

File: deployment/versions.tf (hunk 1 above)
Comment (kirzharov) [reply deployment/versions.tf]: Yes, it was the initial idea 👍 Sorry, my bad, next time I will add a comment about such changes

**Review by scottames — APPROVED** [review]

**Comment by kirzharov** [comment]
@scottames Thanks for the quick review and suggestions! 🙏

**Pull request merged**